numba/tests/test_array_methods.py

```diff
@@ -1780,6 +1782,59 @@ def check(a, b, expected):
         check(arr, arr.T, False)
         check(arr, arr[:-1], False)
 
+    # tests for record  comparison
```

Comment:
I don't think you have to do all of that. Your test should look fairly similar to the `test_identity` a few lines above.